src/wp-includes/webfonts-api/class-wp-webfonts-controller.php

```diff
@@ -0,0 +1,247 @@
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $provider_id Provider ID to fetch.
+	 * @return array[] Registered webfonts.
+	 */
+	public function get_webfonts_by_provider( $provider_id ) {
+		return $this->webfonts_registry->get_by_provider( $provider_id );
+	}
+
+	/**
+	 * Gets the registered webfonts for the given font-family.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $font_family Family font to fetch.
+	 * @return array[] Registered webfonts.
+	 */
+	public function get_webfonts_by_font_family( $font_family ) {
+		return $this->webfonts_registry->get_by_font_family( $font_family );
+	}
+
+	/**
+	 * Gets the registered providers.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return WP_Webfonts_Provider[] Registered providers.
+	 */
+	public function get_registered_providers() {
+		return $this->providers->get_all_registered();
+	}
+
+	/**
+	 * Registers the given provider.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @param string $classname The provider class name.
+	 * @return bool True when registered. False when provider does not exist.
+	 */
+	public function register_provider( $classname ) {
+		return $this->providers->register( $classname );
+	}
+
+	/**
+	 * Generate and enqueue webfonts styles.
+	 *
+	 * @since 5.9.0
+	 */
+	public function generate_and_enqueue_styles() {
+		// Generate the styles.
+		$styles = $this->generate_styles();
+
+		// Enqueue the stylesheet.
+		wp_register_style( $this->stylesheet_handle, '' );
+		wp_enqueue_style( $this->stylesheet_handle );
+
+		// Add the styles to the stylesheet.
+		wp_add_inline_style( $this->stylesheet_handle, $styles );
+	}
+
+	/**
+	 * Generate and enqueue editor styles.
+	 *
+	 * @since 5.9.0
+	 */
+	public function generate_and_enqueue_editor_styles() {
+		wp_add_inline_style( 'wp-block-library', $this->generate_styles() );
+	}
+
+	/**
+	 * Generate styles for webfonts.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return string $styles Generated styles.
+	 */
+	private function generate_styles() {
+		$styles = '';
+		foreach ( $this->get_registered_providers() as $provider_id => $provider ) {
+			$registered_webfonts = $this->webfonts_registry->get_by_provider( $provider_id );
+
+			if ( empty( $registered_webfonts ) ) {
+				continue;
+			}
+
+			add_action( 'wp_head', array( $this, 'render_links' ) );
+
+			$provider->set_webfonts( $registered_webfonts );
+			$styles .= $provider->get_css();
+		}
+		return $styles;
+	}
+
+	/**
+	 * Renders the HTML `<link>` for each provider into `<head>` for enqueued webfonts.
+	 *
+	 * @since 5.9.0
+	 */
+	public function render_links() {
+		echo $this->providers->get_links();
+	}
+
+	/**
+	 * Get the webfonts registry.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return WP_Webfonts_Registry
+	 */
+	public function get_webfonts_registry() {
+		return $this->webfonts_registry;
+	}
+
+	/**
+	 * Get the providers registry.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return WP_Webfonts_Provider_Registry
+	 */
+	public function get_providers() {
+		return $this->providers;
+	}
```

Comment:
@aristath Not a blocker from my end, but particularly for read-only accessing class instances within other class instances, I think the approach without `get_` prefix is reasonable. For example `WP_Customize_Manager` does that too, see its `controls()`, `containers()` etc methods. I personally prefer it because it allows for a shorter and easier-to-read syntax when concatenating method calls. Even more so because there is no (need for) a corresponding setter method.

Reply:
Done in commit https://github.com/WordPress/wordpress-develop/pull/1736/commits/db1eda9b1a5ad589e4038bb7872cc178bdda74af 👍 

These are getters, as they expose a path into the Controller to get an internal property. `get_*` is a common naming to more clearly tell what it does and will return.

As @felixarntz points out, there is precedence in core already for using these shorthand getter naming conventions. And there is also precedence for using the `get_*` getter naming convention.

It's a toss-up to use one or the other. To keep it simple in the global functions, I made the change to the shorthand version.